### PR TITLE
Update the usage of set-output command in GH actions

### PR DESCRIPTION
**Pull request by vasireddy99**
**Description:**

This PR updates the usage of set-output command in GH actions.

Reference : https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Testing:

Test result: https://github.com/vasireddy99/aws-otel-collector/actions/runs/3497301458/jobs/5856226845
Targeted change: https://github.com/vasireddy99/aws-otel-collector/blob/test/tools/release/adot-operator-images-mirror/version-check.sh#L18-L24

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tools/workflow/docker-version-compare.sh [hunk 1]
```diff
@@ -99,9 +99,8 @@ else
...
-echo "::set-output name=minor-update::${MINOR_UPDATE}"
-echo "::set-output name=patch-update::${PATCH_UPDATE}"
-echo "::set-output name=latest-or-newer::${LATEST_OR_NEWER}"
-echo "::set-output name=same-version::${SAME_VERSION}"
-
+echo "major-update=${MAJOR_UPDATE}" >> $GITHUB_OUTPUT
+echo "minor-update=${MINOR_UPDATE}" >> $GITHUB_OUTPUT
+echo "patch-update=${PATCH_UPDATE}" >> $GITHUB_OUTPUT
+echo "latest-or-newer=${LATEST_OR_NEWER}" >> $GITHUB_OUTPUT
+echo "same-version=${SAME_VERSION}" >> $GITHUB_OUTPUT
```
Comment (bryan-aguilar) [inline tools/workflow/docker-version-compare.sh:106]: Same inquires here about the access from a shell script to the $GITHUB_OUTPUT var

File: tools/release/adot-operator-images-mirror/version-check.sh [hunk 2]
```diff
@@ -19,7 +19,7 @@ ECR_TAGS=$(aws ecr-public describe-image-tags --repository-name adot-operator --
 OPERATOR_TAGS=$(curl https://api.github.com/repos/open-telemetry/opentelemetry-operator/tags)
 
 if grep -q "$VERSION" <<< "$OPERATOR_TAGS" && !(grep -q "$VERSION" <<< "$ECR_TAGS"); then
-        echo "::set-output name=update-operator::true"
+        echo "update-operator=true" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (vasireddy99) [reply tools/release/adot-operator-images-mirror/version-check.sh:22]: The testing is done in yaml format, but not in the shell script, have found some good [references](https://github.com/actions-marketplace-validations/whisperity_CodeChecker-Action/commit/149e5f70f9b6aecf846a5b11b9a049ba3f4e0704) in using the env variables for shell script before making this chang

I believe it should be ran under the gh action, not sure if this work in local shell.

[the remark this answers is not in the document]
Comment (vasireddy99) [reply tools/release/adot-operator-images-mirror/version-check.sh:22]: Test result: https://github.com/vasireddy99/aws-otel-collector/actions/runs/3497301458/jobs/5856226845
Targeted change: https://github.com/vasireddy99/aws-otel-collector/blob/test/tools/release/adot-operator-images-mirror/version-check.sh#L18-L24

[the remark this answers is not in the document]
Comment (bryan-aguilar) [reply tools/release/adot-operator-images-mirror/version-check.sh:22]: Thank you for showing a test workflow. Could you expand on your test workflow by accessing that output from the shell script in a different GHA workflow job? Currently the test workflow only shows that there was no error and not that the output value is accessible.

**Review by PaurushGarg — APPROVED** [review]
Looks good to me.

File: tools/release/adot-operator-images-mirror/version-check.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (vasireddy99) [reply tools/release/adot-operator-images-mirror/version-check.sh:22]: That is a goood point. i have modified the [ workflow ](https://github.com/vasireddy99/aws-otel-collector/blob/test/.github/workflows/CD-adot-operator.yml#L59) to test whether the value is set in `GITHUB_OUPUT`. echoed, retrieved the value of update-operator as [expected](https://github.com/vasireddy99/aws-otel-collector/actions/runs/3879140562/jobs/6616030976#step:7:1)

**Pull request merged**